samcli/lib/config/samconfig.py

```diff
@@ -39,11 +40,18 @@ def __init__(self, config_dir, filename=None):
             Optional. Name of the configuration file. It is recommended to stick with default so in the future we
             could automatically support auto-resolving multiple config files within same directory.
         """
+        self.document = {}
         self.filepath = Path(config_dir, filename or DEFAULT_CONFIG_FILE_NAME)
+        self.file_manager = self.FILE_MANAGER_MAPPER.get(self.filepath.suffix[1:], None)  # default to TOML
+        if not self.file_manager:
+            LOG.warning(f"The config file extension '{self.filepath.suffix[1:]}' is not supported.")
+            raise SamConfigFileReadException(
+                f"The config file {self.filepath} uses an unsupported extension, and cannot be read."
```

Comment:
Should we also list the supported extensions for users to see what is supported?

Reply:
We could, though if the number of supported formats grows, it could be a large list. But for now with 3 (4 if you want to count .yml and .yaml as separate) supported formats, it would probably be a good idea yes. I'll add that in 👍